### PR TITLE
Smell documentation link

### PR DESCRIPTION
**Pull request by danielmbarlow**
Added links on smells to Reek documentation on Github in the code screen. All comments are welcome. We don't use Minitest at work, so let me know if there's anything you want me to change with the tests.

### CODE REVIEW

File: lib/rubycritic/core/smell.rb [hunk 1]
```diff
@@ -46,6 +46,14 @@ def to_json(*a)
       to_h.to_json(*a)
     end
 
+    def dasherized_type
```
Comment (troessner) [inline lib/rubycritic/core/smell.rb]: Lets make this method private, it doesnt have to be public.

File: test/lib/rubycritic/core/smell_test.rb [hunk 2]
```diff
@@ -69,4 +69,27 @@
       smell1.must_equal smell2
     end
   end
+
+  describe '#dasherized_type' do
```
Comment (troessner) [inline test/lib/rubycritic/core/smell_test.rb]: I believe we should test this method implicitly through doc_url  - how about moving those tests there?

**Comment by troessner** [comment]
Btw. this functionality (obviously) exists in Reek as well, unfortunately it's not really well exposed:

```Ruby
warning = FactoryGirl.build(:smell_warning)
Reek::Report::WikiLinkWarningFormatter.new.send :explanatory_link, warning

>>
=> "https://github.com/troessner/reek/blob/master/docs/Feature-Envy.md"

It probably would make sense to expose this in a better way. But that would require a little more sophisticated approach than just making the method public ;)

I guess for now it's better to go with your solution.
Two small comments and the lgtm

**Comment by danielmbarlow** [comment]
Hi Timo, I made the changes you suggested, squashed my commits (as I think your CONTRIBUTE.md suggests), and rebased on master. Let me know if you need anything else done for this PR.

**Comment by troessner** [comment]
Looks good now!  But travis rightfully complains about rubocop violations ;)

**Comment by danielmbarlow** [comment]
Hi Timo, how are we looking now?

**Comment by troessner** [comment]
Looks good now! travis is still red because of jruby but that shouldn't be an issue after https://github.com/whitesmith/rubycritic/pull/128 -> merged!

**Comment by troessner** [comment]
Btw. extremely awesome pull request, a small change but one that increases convenience significantly, thanks for that!

**Comment by danielmbarlow** [comment]
You are welcome. I look forwards to using it. Thanks for Rubycritic!

**Comment by Onumis** [comment]
This is so awesome! Thanks @danielmbarlow

**Pull request merged**